addons/mail/static/src/messaging/entity/partner/partner.js

```diff
@@ -202,6 +218,17 @@ function PartnerFactory({ Entity }) {
             }, 50 * 1000);
         }
 
+        /**
+         * @private
+         * @returns {integer}
+         */
+        _computeId() {
+            if (this.isPublic && (!this.id || this.id > 0)) {
+                return getNextPublicId();
+            }
+            return this.id;
+        }
```

Comment:
Each fake partner has its own name and country, hence they cannot share a same entity `publicPartner`.

Reply:
I don't like that this is a compute depending on `isPublic`... If the value of `isPublic` is changed for whatever reason, it will generate a new id...

Reply:
This compute is necessary in order to auto-generate partner ids that have `isPublic` set. These are no true partner, in fact they can't be uniquely identified from server data, which explains why we have to auto-compute locally.

All this is workaround from lack of traceability of such public partners from server, and this compute Id is a workaround to trace these partners in JS. At the moment, there is no reason to change `isPublic` on partners, and there is no motivation to change that until either server tracks them properly (or we replace them by genuine partners, if we want later support case of livechat session that started with unlogged visitor and he/she becomes logged during livechat session at some point in time).

Unfortunately I don't think we can do much better until then.

Reply:
This fake partner is likely flux specific then and should probably be handled/created by the caller (notification handler) instead of by the partner entity itself with a compute.

If we somehow allow "reconciliation", the current partner would make the localId become unsync with the id itself? How about relations and _findFunctionFromData?

I think this is too dangerous.

Reply:
Move fake id into convert data of livechat, remove is public attr